### PR TITLE
WIP: Add ability to hide "Download" for a video

### LINKED ISSUE #536: Add ability to hide "Download" for a video
Some uploaders don't want people to be able to easily download their video. It's their choice, so let's add an opt-out checkbox in the upload form.

Of course, this is just "cosmetic". If someone really wants to download the video, he will be able to do it (like we can on other video platforms with youtube-dl for example).

### PR DESCRIPTION
**Pull request by lucas-dclrcq**
fixes #536

### CODE REVIEW

File: client/src/app/videos/+video-edit/shared/video-edit.component.html [hunk 1]
```diff
@@ -125,6 +125,11 @@
               i18n-labelText labelText="Enable video comments"
             ></my-peertube-checkbox>
 
+            <my-peertube-checkbox
+                    inputName="downloadingEnabled" formControlName="downloadingEnabled"
```
Comment (Chocobozzz) [inline client/src/app/videos/+video-edit/shared/video-edit.component.html]: only 2 spaces for the indentation

File: server/controllers/api/videos/index.ts [hunk 2]
```diff
@@ -179,6 +179,7 @@ async function addVideo (req: express.Request, res: express.Response) {
     licence: videoInfo.licence,
     language: videoInfo.language,
     commentsEnabled: videoInfo.commentsEnabled || false,
+    downloadingEnabled: videoInfo.downloadingEnabled || false,
```
Comment (Chocobozzz) [inline server/controllers/api/videos/index.ts]: `|| true`

File: server/tools/peertube-upload.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ if (!program['tags']) program['tags'] = []
 if (!program['nsfw']) program['nsfw'] = false
 if (!program['privacy']) program['privacy'] = VideoPrivacy.PUBLIC
 if (!program['commentsEnabled']) program['commentsEnabled'] = false
+if (!program['downloadingEnabled']) program['downloadingEnabled'] = false
```
Comment (Chocobozzz) [inline server/tools/peertube-upload.ts]: `true`

File: server/helpers/custom-validators/activitypub/videos.ts [hunk 4]
```diff
@@ -67,6 +67,7 @@ function sanitizeAndCheckVideoTorrentObject (video: any) {
     isVideoViewsValid(video.views) &&
     isBooleanValid(video.sensitive) &&
     isBooleanValid(video.commentsEnabled) &&
+    isBooleanValid(video.downloadingEnabled) &&
```
Comment (Chocobozzz) [inline server/helpers/custom-validators/activitypub/videos.ts]: Add `if (!isBooleanValid(video.downloadingEnabled)) video.downloadingEnabled = true` for backward compatibility

**Comment by lucas-dclrcq** [comment]
Thanks for the review ! I'll add unit tests as requested :)

**Comment by Chocobozzz** [comment]
@lucas-dclrcq Any news on this PR? Do you need help?

**Comment by lucas-dclrcq** [comment]
Sorry, I was quite busy this last few weeks, I'll try to finish it this weekend.

**Comment by Chocobozzz** [comment]
Merged manually. Thanks @lucas-dclrcq

**Pull request merged**